### ISSUE TITLE
[v0.14-beta3] Quick tip does nothing

### ISSUE THREAD

**Issue by rapus95** [julia-vscode/julia-vscode#971]
MWE:
```julia
struct DimensionalProduct{P,M,N}
    DimensionalProduct{M,N}() where {M,N} = new{M*N,M,N}()
end
```
shows a quick tip to add a default constructor. But on clicking on it (tried it out of curiosity) nothing happened. Quick tip still showing up all time. (At least on beta 2 and beta 3)

**Comment by ZacLN** [comment]
In this case you've already got a constructor defined within the struct definition so nothing new is added. 

Quick tips' availability are calculated based on the context of the cursor (within a struct definition), only after clicking on them do we try to work out whether they're appropriate. The behaviour could be altered so a change is made anyway but that seems less intuitive to me.

**Comment by davidanthoff** [comment]
> only after clicking on them do we try to work out whether they're appropriate

Could we change that so that they are only offered if they are appropriate?

**Comment by ZacLN** [comment]
Not really, they typically involve a fair bit of calculation. I'm pretty sure this approach with consistent with how it's handled for the typescript extension

**Comment by davidanthoff** [comment]
For now just moving off the milestone, doesn't seem critical.

**Comment by rapus95** [comment]
I guess disabling the default constructor case would be the better approach for now:
1. When there already exists a constructor it doesn't do anything.
2. When there are no attributes (Singleton type) it doesn't do anything.
3. If there is no constructor & there are attributes, it's buggy:
```julia
struct MedianAlgo <: MyAbstractAlgo
    f::Int
    function MedianAlgo(args...)

        new(f)
    end
end
```

**Comment by rapus95** [comment]
If you can't check the applicability before showing the icon you might consider moving the bulb to its own column to guarantee it to always show up on the same line as the cursor (rather than varying to overcome missing leading spaces). Maybe resize the bulb to make it smaller and instead show it in every place where applicable.

Right now it is a big distraction tbh. It somehow interrupts the code flow IMO as it lags behind the cursor and thus somewhat keeps catching the focus of the user.

Eclipse IMO solved that very good. See: https://www.youtube.com/watch?v=IZBCGtY00y0

**Comment by davidanthoff** [comment]
Hm, maybe we should remove it for now?

**Comment by ZacLN** [comment]
Tbh I don't think it's a distraction. Either way the first hit of a google search tells you how to turn it [off](https://stackoverflow.com/questions/39485506/how-to-disable-the-light-bulb-in-visual-studio-code)

**Comment by rapus95** [comment]
Either way, this issue should stay open as the quick tip is still buggy.

**Comment by davidanthoff** [comment]
If there is something that is actually not working, we should fix it, but I agree, lets leave the functionality in there.

@ZacLN do we have enough info from @rapus95 to pin down the non-working case?

**Comment by ZacLN** [comment]
Addressing the 3 points raised above: 1 and 2 are decisions and not bugs, not clear what the issue with the example in 3?

**Comment by rapus95** [comment]
Well, I guess I have a weird perspective here, but why would you decide to do nothing in any case? The only situation where people click on the quick tip is if they want to add a default constructor. No matter whether there are already constructors or not. Thus, they want something to happen. Deciding to purposefully do nothing is like redering the functionality effectively useless and baiting. Why would you even show it if you don't intend to do what the user wants it to do? There's no need to protect the user from accidental clicks on it. Thus, whenever he clicks it, he want's something to happen. Not nothing.
And for the 3rd case, the default that is generated produces a compile error as it is right now. Watch the different symbols in the argument vs the ones in `new`.

**Comment by davidanthoff** [comment]
Should we move to the v0.14.1 milestone so that we can ship? Still not clear to me whether there is even a TODO here for us :)

**Comment by rapus95** [comment]
What's not clear?
```julia
function Name(args....)

    new(f)
end
```
won't compile. And not doing anything when a user clicks on "add default constructor" is definitively counter-intuitive at any time.
In all cases I'd do exactly what the user requested:
```julia
function Name(arg1, ..., argn)
    #TODO insert custom code
    new(arg1, ..., argn)
end
```
That's the default constructor.
